frontends/ntt-pipeline/gen-ntt-pipeline.py

```diff
@@ -207,10 +207,12 @@ def preamble_group(comp: cb.ComponentBuilder, row):
         with main.group(f"preamble_{row}") as preamble:
             input.addr0 = row
             phis.addr0 = row
-            reg.write_en = 1
-            reg.in_ = input.read_data
-            phi.write_en = 1
-            phi.in_ = phis.read_data
+            input.content_en = 1
+            phis.content_en = 1
+            reg.write_en = input.done @ 1
+            reg.in_ = input.done @ input.read_data
+            phi.write_en = input.done @ 1
+            phi.in_ = phis.done @ phis.read_data
             preamble.done = (reg.done & phi.done) @ 1
```

Comment:
@KabirSamsi this is a case of [guard conjunction](https://docs.calyxir.org/lang/ref.html#guards) in the wild. @rachitnigam I'm guessing looks pretty risky to you haha

Reply:
Yup! Pretty risky. I would restructure this as two static groups that write to the memory and run in parallel instead.

Reply:
Alrighty, there is now https://github.com/calyxir/calyx/pull/2131

Reply:
Should we keep reviewing this PR, or review #2131 instead?

Reply:
You can focus on this one Kabir, but we'll keep both PRs! There will be a minor merge conflict when one of these two is merged into main and the other tries to merge into the new main, but I'll deal with that.

Reply:
In truth maybe I should have branched off this branch, and chosen this branch as the base of the merge, for 2131. Ah well. My bad, and I'll eat the merge conflict issues

Reply:
The other branch has been merged into main and this one has been caught up to those changes. Ready for a final look. Once you have completed the review process, Kabir, I'll hit merge.